File: 1.1.py
import random

# Generar el vector de forma aleatoria
vector = [random.randint(100, 999) for _ in range(20)]

# Mostrar el vector generado
print("Vector generado aleatoriamente:")
print(vector)

# a. Encontrar el menor valor y contar cuántas veces se repite
min_value = min(vector)
min_count = vector.count(min_value)
print("\nMenor valor en el vector:", min_value)
print("Se repite", min_count, "veces.")

# b. Encontrar el mayor valor y sus posiciones
max_value = max(vector)
max_positions = [i for i, x in enumerate(vector) if x == max_value]
print("\nMayor valor en el vector:", max_value)
print("Se encuentra en la(s) posición(es):", max_positions)

File: 1.1.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#define TAM 20

void CargaAleatoria(int[], int, int);
int Minimo(int[], int);
int Maximo(int[], int);
void MostrarMinimo(int[], int, int);
void MostrarMaximo(int[], int, int);
/*
a. Mostrar el menor valor e indicar cuantas veces se repite
b. Mostrar el mayor valor e indicar en que posición/posiciones se encuentra
*/

int main()
{
    int vec[20], min, max;

    srand(time(NULL));              
    CargaAleatoria(vec, TAM, 1000);
    min= Minimo(vec, TAM);
    max= Maximo(vec, TAM);
    MostrarMinimo(vec, min, TAM);
    MostrarMaximo(vec, max, TAM);   
    return 0;
}

// N= Cantidad de Elementos
void CargaAleatoria(int vec[], int N, int max)
{
    int i;
    for(i=0; i<N; i++)
    {
        vec[i]=rand()%max;
    }

}

int Minimo(int vec[], int N)
{
    int min=vec[0], i;
    for(i=1; i<N; i++)
    {
        if(vec[i]<min)
            min=vec[i];
    }
    return min;
}

int Maximo(int vec[], int N)
{
    int max=vec[0], i;
    for(i=1; i<N; i++)
    {
        if(vec[i]>max)
            max=vec[i];
    }
    return max;
}

void MostrarMinimo(int vec[], int min, int N)
{
    int contador=0, i;
    
    for(i=0; i<N; i++)
    {
        if(vec[i]==min)
            contador++;
    }
    printf("\nEl valor minimo es: %d y se repite %d veces.", min, contador);
    
}

void MostrarMaximo(int vec[], int max, int N)
{
    int i;
    printf("El valor maximo es: %d y se encuentra en las siguientes posiciones del vector:\n", max);
    for(i=0; i<N; i++)
    {
        if(vec[i]==max)
            printf("%d\n", i+1);
    }
}